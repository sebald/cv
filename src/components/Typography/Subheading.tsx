import {
  fontWeight,
  suit,
  textAlign,
  textTransform,
  typography,
  StyledProps,
} from '../../style';

export const Subheading = suit<
  StyledProps<'fontWeight' | 'textAlign' | 'textTransform'>
>('h3')`
  ${typography('subheading')};
  ${fontWeight};
  ${textTransform};
  ${textAlign};
`;
Subheading.displayName = 'Subheading';
