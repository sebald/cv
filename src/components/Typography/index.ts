export * from './Caption';
export * from './Heading';
export * from './Display';
export * from './Subheading';
export * from './Text';
