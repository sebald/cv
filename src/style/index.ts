export { default as theme, Theme } from './theme';

export * from './mixins';
export * from './styled-components';
export * from './utils';
