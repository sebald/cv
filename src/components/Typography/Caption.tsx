import { styled, typography } from '../../style';

export const Caption = styled.h5`
  ${typography('caption')};
`;
Caption.displayName = 'Caption';
