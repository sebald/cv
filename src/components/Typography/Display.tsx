import { styled, typography } from '../../style';

export const Display = styled.h1`
  ${typography('display')};
`;
Display.displayName = 'Display';
