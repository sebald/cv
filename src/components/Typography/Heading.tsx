import { styled, typography } from '../../style';

export const Heading = styled.h2`
  ${typography('heading')};
`;
Heading.displayName = 'Heading';
