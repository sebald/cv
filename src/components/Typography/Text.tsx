import { styled, typography } from '../../style';

export const Text = styled.p`
  ${typography('body')};
`;
Text.displayName = 'Text';
