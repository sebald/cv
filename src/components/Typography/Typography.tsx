import capitalize from 'lodash-es/capitalize';
import {
  fontWeight,
  suit,
  textAlign,
  textTransform,
  typography,
  StyledProps,
  TTag,
} from '../../style';
import { TypographyStyle } from '../../style/theme';

type TypographyProps = StyledProps<
  'fontWeight' | 'textAlign' | 'textTransform'
>;

const createTypography = (name: TypographyStyle, tag: TTag) => {
  const Component = suit<TypographyProps>(tag)`
    ${typography(name)};
    ${fontWeight};
    ${textTransform};
    ${textAlign};
  `;
  Component.displayName = capitalize(name);
  return Component;
};

export const Display = createTypography('display', 'h1');
export const Heading = createTypography('heading', 'h2');
export const Subheading = createTypography('subheading', 'h3');
export const Caption = createTypography('caption', 'h6');
export const Text = createTypography('body', 'p');
