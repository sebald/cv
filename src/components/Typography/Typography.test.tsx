import { shallow } from 'enzyme';
import * as React from 'react';

import { theme } from '../../style';
import { Caption, Display, Heading, Subheading, Text } from './Typography';

test('render display', () => {
  const wrapper = shallow(<Display theme={theme}>Text</Display>);
  expect(wrapper).toMatchSnapshot();
});

test('render heading', () => {
  const wrapper = shallow(<Heading theme={theme}>Text</Heading>);
  expect(wrapper).toMatchSnapshot();
});

test('render subheading', () => {
  const wrapper = shallow(<Subheading theme={theme}>Text</Subheading>);
  expect(wrapper).toMatchSnapshot();
});

test('render caption', () => {
  const wrapper = shallow(<Caption theme={theme}>Text</Caption>);
  expect(wrapper).toMatchSnapshot();
});

test('render text', () => {
  const wrapper = shallow(<Text theme={theme}>Text</Text>);
  expect(wrapper).toMatchSnapshot();
});
