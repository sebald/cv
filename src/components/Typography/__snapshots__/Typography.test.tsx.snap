// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render caption 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 2rem;
  padding-top: 0.625rem;
  margin-top: 0;
  margin-bottom: 1.375rem;
  color: #9e9e9e;
  font-family: 'Montserrat',sans-serif;
  font-weight: 300;
}

<h6
  className="c0"
>
  Text
</h6>
`;

exports[`render display 1`] = `
.c0 {
  font-size: 3.375rem;
  line-height: 4rem;
  padding-top: 0.75rem;
  margin-top: 0;
  margin-bottom: 1.25rem;
  color: #212121;
  font-family: 'Montserrat',sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
}

<h1
  className="c0"
>
  Text
</h1>
`;

exports[`render heading 1`] = `
.c0 {
  font-size: 2.25rem;
  line-height: 4rem;
  padding-top: 0.25rem;
  margin-top: 0;
  margin-bottom: 1.75rem;
  color: #212121;
  font-family: 'Montserrat',sans-serif;
  font-weight: 600;
  text-transform: uppercase;
}

<h2
  className="c0"
>
  Text
</h2>
`;

exports[`render subheading 1`] = `
.c0 {
  font-size: 1.5rem;
  line-height: 2rem;
  padding-top: 0.4375rem;
  margin-top: 0;
  margin-bottom: 1.5625rem;
  color: #212121;
  font-family: 'Montserrat',sans-serif;
  font-weight: 600;
  text-transform: uppercase;
}

<h3
  className="c0"
>
  Text
</h3>
`;

exports[`render text 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 2rem;
  padding-top: 0.625rem;
  margin-top: 0;
  margin-bottom: 1.375rem;
  color: #424242;
  font-family: 'Raleway',sans-serif;
  font-weight: 300;
}

<p
  className="c0"
>
  Text
</p>
`;
